listenbrainz/messybrainz/transfer_to_timescale.py

```diff
@@ -0,0 +1,108 @@
+"""
+Steps to move MsB to TS database.
+
+    1. Create the destination tables and index. The index only exists to avoid duplicates which may happen
+       when the script is run multiple times.
+    2. Run the script for first time.
+    3. Stop TS writer.
+    4. Rerun the script to insert submissions since first run.
+    5. Switch MsB submission to new tables by updating TS writer.
+    6. Profit!
+"""
+
+from flask import current_app
+from psycopg2.extras import execute_values
+from sqlalchemy import text
+
+from listenbrainz import messybrainz
+from listenbrainz.db import timescale
+
+
+def retrieve_data(last_row_id):
+    query = """
+            SELECT r.id
+                 , r.gid AS recording_msid
+                 , rj.data->>'title' AS recording
+                 , rj.data->>'artist' AS artist
+                 , rj.data->>'release' AS release
+                 , rj.data->>'track_number' AS track_number
+                 , r.submitted
+              FROM recording r
+              JOIN recording_json rj
+                ON r.data = rj.id
+             WHERE r.id > :last_row_id
+          ORDER BY r.id ASC
+        FETCH NEXT 50000 ROWS ONLY
+    """
+    with messybrainz.engine.connect() as msb_conn:
+        results = msb_conn.execute(text(query), last_row_id=last_row_id)
+        return results.fetchall()
+
+
+def insert_data(values):
+    raw_conn = timescale.engine.raw_connection()
+    query = """
```

Comment:
I love that this comment is SQL, not python. :)

Reply:
:)